packages/shared/src/components/plus/PlusExtension.tsx

```diff
@@ -0,0 +1,92 @@
+import type { ReactElement } from 'react';
+import React, { useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { PlusUser } from '../PlusUser';
+import { TypographyType } from '../typography/Typography';
+import { IconSize } from '../Icon';
+import { PlusList } from './PlusList';
+import { plusRedBackgroundImage } from '../../lib/image';
+import { Image } from '../image/Image';
+import { Button, ButtonVariant } from '../buttons/Button';
+import { webappUrl } from '../../lib/constants';
+import { PlusInfo } from './PlusInfo';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent, Origin, TargetType } from '../../lib/log';
+import { MarketingCtaVariant } from '../marketingCta/common';
+import { useBoot } from '../../hooks';
+import { getPricePreviews } from '../../graphql/paddle';
+import { PlusPriceTypeAppsId } from '../../lib/featureValues';
+
+const PlusExtension = (): ReactElement => {
+  const { getMarketingCta } = useBoot();
+  const marketingCta = getMarketingCta(MarketingCtaVariant.Plus);
+  const { flags } = marketingCta;
+  const { logEvent } = useLogContext();
+  const { data: productOptions } = useQuery({
+    queryKey: generateQueryKey(RequestKey.PricePreview),
+    queryFn: async () => {
+      const previews = await getPricePreviews();
+      return previews.filter(
+        (item) => item.appsId !== PlusPriceTypeAppsId.GiftOneYear,
+      );
+    },
+  });
```

Comment:
Why does this one use a raw query and the web one the context?
Can't this use the context?

Reply:
Ah ok, figured it was something like that.
But the filter I saw somewhere before, can we maybe abstract that into a function somehow?